networking.md

```diff
@@ -0,0 +1,463 @@
+        enabled: false
+      name: enp6s0.20
+      state: up
+      type: vlan
+      vlan:
+        base-iface: enp6s0
+        id: 20
+    - description: storage vlan interface
+      ipv4:
+        address:
+        - ip: 172.18.0.10
+          prefix-length: 24
+        enabled: true
+      ipv6:
+        enabled: false
+      name: enp6s0.21
+      state: up
+      type: vlan
+      vlan:
+        base-iface: enp6s0
+        id: 21
+    - description: tenant vlan interface
+      ipv4:
+        address:
+        - ip: 172.19.0.10
+          prefix-length: 24
+        enabled: true
+      ipv6:
+        enabled: false
+      name: enp6s0.22
+      state: up
+      type: vlan
+      vlan:
+        base-iface: enp6s0
+        id: 22
+    - description: Configuring enp6s0
+      ipv4:
+        address:
+        - ip: 192.168.122.10
+          prefix-length: 24
+        enabled: true
+      ipv6:
+        enabled: false
+      mtu: 1500
+      name: enp6s0
+      state: up
+      type: ethernet
+  nodeSelector:
+    kubernetes.io/hostname: worker-0
+    node-role.kubernetes.io/worker: ""
+```
+
+### Network attachment definitions
+For each isolated network, to which a service deployment pod should be attached, a `network-attachment-definition` (`nad`) need to be created.
+
+This example `nad` for the internalapi network of type `macvlan` use the previous created `enp6s0.20` interface and the `whereabouts` CNI IPAM plugin to assign IPs to the created pods from the range `.30` - `.70`.
+
+* internalapi network-attachment-definition
+```yaml
+apiVersion: k8s.cni.cncf.io/v1
+kind: NetworkAttachmentDefinition
+metadata:
+  name: internalapi
+  namespace: openstack
+spec:
+  config: |
+    {
+      "cniVersion": "0.3.1",
+      "name": "internalapi",
+      "type": "macvlan",
+      "master": "enp6s0.20",
+      "ipam": {
+        "type": "whereabouts",
+        "range": "172.17.0.0/24",
+        "range_start": "172.17.0.30",
+        "range_end": "172.17.0.70"
+      }
+    }
+```
+
+* storage network-attachment-definition
+```yaml
+apiVersion: k8s.cni.cncf.io/v1
+kind: NetworkAttachmentDefinition
+metadata:
+  name: storage
+  namespace: openstack
+spec:
+  config: |
+    {
+      "cniVersion": "0.3.1",
+      "name": "storage",
+      "type": "macvlan",
+      "master": "enp6s0.21",
+      "ipam": {
+        "type": "whereabouts",
+        "range": "172.18.0.0/24",
+        "range_start": "172.18.0.30",
+        "range_end": "172.18.0.70"
+      }
+    }
+```
+
+* [macvlan plugin](https://www.cni.dev/plugins/current/main/macvlan/)
+* [whereabouts IPAM CNI plugin - Extended configuration](https://github.com/k8snetworkplumbingwg/whereabouts/blob/master/doc/extended-configuration.md)
+
+### MetalLB IP address pool
+On the isolated network where a service need to expose its endpoint using MetalLB, an `IPAddressPool` needs to be specified, which specifies at least the addresses should be used when a VIP gets requested, in this example `.80` - `.90`:
```

Comment:
I don't understand what you mean by "`which specifies at least the addresses should be used when a VIP gets requested`"

Reply:
let me know what you think about the current version. just used parts of the ipaddresspool crd description

Reply:
Looks good